tests/scripts/travis

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+. tests/scripts/history_checks.bash
+
+#### Utilities
+function msg() {
+    echo "travis_tests: $*"
+}
+
+function err() {
+    echo "travis_tests: $*" 1>&2
+}
+
+####
+# If in travis context we need to checkout a branch, otherwise travis works in detached head.
+if ! git checkout -b travis_tests
+then
+    err "failed to create a branch"
+    exit 1
+fi
+
+####
+msg "Check presence of binary files in branch history."
+branch_res=0
+
+if ! check_branch # Read: "if check fails"
+then
+    err "ERROR: something went wrong in branch checks"
+    branch_check=1
+fi
+
+####
+msg "Check code style."
+style_res=0
+
+if ! make style-check
+then
```

Comment:
Maybe the style check was not the best first example (we can discuss in person), but it seems to me adding https://mybitbucket.llnl.gov/projects/MFEM/repos/tests/browse/gitignore should be straightforward here.

It also related because it will help with avoid accidental commit of the executables.